### PR TITLE
Rm nightly feature probing in `build.rs`

### PR DESCRIPTION
**Pull request by NobodyXu**
People who need this can simply pass `--cfg theerror_error_generic_member_access` to enable it.

### CODE REVIEW

**Review by Byron — APPROVED** [review]
This looks like a change that will accelerate the initial build measurably! A great catch!

----

By the way, the way I see it, you are the driver of all of this, and I am at best a spectator who is along for the ride who can provide a specific, higher-level perspective. Maybe this is something that should be represented in the ownership of this repository and the crate as well - I'd be happy to hand it over. Nothing has to change as I can still take a look at PRs, but I'd see them more from a perspective of usability and maybe compatibility, while not being able to support on a code-review level at all. To me, proc-macros seems like an entirely different language and I am lacking the experience and mental model to really know what's going on.

In other words, in theory you could push directly to `main` for a zero-fuzz development while the crate isn't used by `gitoxide` or anyone else, and I would primarily engage via issues or design discussions if there even is anything to discuss or flesh out.

Please let me know what you have in mind in terms of collaboration so we can align on a workflow.
Thank you.

PS: I also renamed this repository to match the name used in the cargo manifest.

**Comment by NobodyXu** [comment]
@Byron That sounds good to me, I will merge simple PRs like this andnon-functional changes directly, but for anything related to API/usage/design, I will still ask you to review it.

**Comment by Byron** [comment]
Thanks, let's do that!

And when `therror` gets the attention it deserves assuming it does turn out as (r)evolutionary as I think it will be, I will ask you again if you'd like to take full ownership :) - after all I am really just the one who reserved the name in anticipation, who did the groundbreaking.
